[Merged by Bors] - Push metrics to remote server

valar999 opened the pull request:
## Motivation

To get metrics from external nodes we have to add push mechanism.

Part of https://github.com/spacemeshos/pm/issues/37

## Changes

- adds flag `--metrics-push` to specify remote server url
- `metrics-push-period` specify frequency of pushing

## Test Plan

Test infrastructure is ready and receiving of metrics was tested.

## DevOps Notes
<!-- Please uncheck these items as applicable to make DevOps aware of changes that may affect releases -->
- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

valar999 committed:
Add a flag to push prometheus metrics

valar999 committed:
Fix file name in hare/metrics

valar999 committed:
go mod tidy

valar999 committed:
Add comment for StartPushMetrics()

valar999 committed:
Add small change in tests

y0sher commented on cmd/node/node.go:
```diff
@@ -1072,6 +1073,12 @@ func (app *SpacemeshApp) Start(*cobra.Command, []string) {
 		metrics.StartCollectingMetrics(app.Config.MetricsPort)
```
this code starts a regular Promethues client (server for pulling the data from the node). shouldn't we NOT start it when using push metrics?

y0sher replied:
might need a rename to "StartMetricsServer" because apparently it doesn't actually collects the metrics but just exposes the server.

valar999 replied:
> this code starts a regular Promethues client (server for pulling the data from the node). shouldn't we NOT start it when using push metrics?

Metrics push is a separate component that is triggered by a flag. It is intended for collecting metrics from community nodes, and prometheus client collect metrics from spacecraft nodes so they can work in parallel with the prometheus client.

y0sher replied:
I meant renaming the Pull component "MetricsServer" since this one is pushing and ain't a server.

valar999 replied:
You are right.

We have a server to pull metrics from, function that start it is called `StartCollectingMetrics`. It could be renamed to `StartMetricsServer`.

Pushing metrics starts with `StartMetricsServer`. Let's rename it to `StartPushingMetrics`.

valar999 committed:
Rename StartCollectingMetrics to StartMetricsServer

It's a better name for this.

valar999 committed:
Merge branch 'develop' into metrics-push

valar999 committed:
Rename StartCollectingMetrics and StartMetricsServer

StartMetricsServer -> StartPushingMetrics
it pushes metrics

StartCollectingMetrics -> StartMetricsServer
it exposes metrics via http

valar999 committed:
Update go.sum

valar999 commented:
bors merge

valar999 commented:
bors merge

valar999 commented:
bors merge
